.travis.yml

```diff
@@ -32,30 +40,41 @@ addons:
   coverity_scan:
     project:
       name: vcdMaker
-      version: v3.0
+      version: v4.0
       description: Coverity Scan by Travis CI
     notification_email: vcdmaker@mail.com
     build_command_prepend: "cov-configure --comptype gcc --compiler g++-8 --template && cmake -DUSE_STATIC_PUGIXML=ON -DCMAKE_BUILD_TYPE=Release ."
     build_command: "make"
     branch_pattern: coverity_scan
 
-# Blocklist - branches not to be built. RegEx is allowed.
-branches:
-  except:
-  - r2_release
+install:
+  - if [env(CODE_COVERAGE) = 1]; then gem install coveralls-lcov ; fi
+  - if [env(CODE_COVERAGE) = 1]; then apt-get -y install lcov ; fi
 
 # Safelist - branches to be built.
 branches:
   only:
   - master
   - v3_release
+  - v4_release
   - coverity_scan
 
 before_script:
   - git submodule update --init --recursive
   - cd ./sources
+  - if [env(CODE_COVERAGE) = 1]; then mkdir CodeCoverage ; fi
+  - if [env(CODE_COVERAGE) = 1]; then lcov --directory . --zerocounters ; fi
+
 
 script:
-  - cmake -DUSE_STATIC_PUGIXML=ON -DCMAKE_BUILD_TYPE=Release .
-  - VERBOSE=1 make
-  - make check
+  - if [ ${COVERITY_SCAN_BRANCH} != 1 ]; then cmake $OPTIONS ; fi
+  - if [ ${COVERITY_SCAN_BRANCH} != 1 ]; then VERBOSE=1 make ; fi
+  - if [ ${COVERITY_SCAN_BRANCH} != 1 ]; then make check ; fi
+
+after_success:
+  - if [env(CODE_COVERAGE) = 1]; then find CMakeFiles -type f -name "*.gc*" -exec mv {} CodeCoverage \; ; fi
+  - if [env(CODE_COVERAGE) = 1]; then lcov --directory CodeCoverage --capture --output-file coverage.info ; fi
+  - if [env(CODE_COVERAGE) = 1]; then lcov --remove coverage.info '*/unitTests/*' '/usr/*' --output-file coverage.info ; fi
+  - if [env(CODE_COVERAGE) = 1]; then lcov --list coverage.info ; fi
+  - if [env(CODE_COVERAGE) = 1]; then coveralls-lcov --repo-token ${COVERALLS_REPO_TOKEN} coverage.info ; fi
```

Comment:
coveralls-lcov looks really old and unmaintained. And it needs lcov.
Isn't this what you want to use: https://github.com/eddyxu/cpp-coveralls ? This is what they advertise on their website.

Reply:
That was my initial idea. I gave it a try on a local Linux machine and it didn't work. It was failing to parse gcov output. At some point I felt discouraged to use coveralls at all. That's why I used coveralls-lcov. Maybe once cpp-coveralls gets improved we could switch to it.

Reply:
Let's go with lcov for now. I couldn't make it run locally, I mean cpp-coveralls. The bugs listed in the project are manifesting their presence for me as well. Hopefully they will be fixed in the future and we will be able to switch to cpp-coveralls.